File: JavaScript/2-lambda.js
'use strict';

const { log: ln } = Math;
const log = (base, n) => ln(n) / ln(base);

// Usage
{
  const lg = (n) => log(10, n);
  const ln = (n) => log(Math.E, n);

  console.log('lg(5) =', lg(5));
  console.log('ln(5) =', ln(5));
}

File: Java/src/2-lambda.java
import java.util.function.Function;

class Lambda {
    public static void main(String[] args) {
        Function<Double,Double> lg = (Double n) -> log(10d, n);
        Function<Double,Double> ln = (Double n) -> log(Math.E, n);
        //usage
        System.out.println(lg.apply(1000d));
        System.out.println(ln.apply(Math.E*Math.E));
    }

    private static Double log(Double base, Double n)
    {
        return Math.log(n)/ Math.log(base);
    }
}
